File: Desktop/exchange/src/main/java/com/kjb04/exchange/Authentication.java
package com.kjb04.exchange;

import java.util.prefs.Preferences;

public class Authentication {
    private static Authentication instance;
    private static final String TOKEN_KEY = "TOKEN";
    private static final String USERNAME_KEY = "USERNAME";

    private String token;
    private String username;
    private Preferences pref;
    private Authentication() {
        pref = Preferences.userRoot().node(this.getClass().getName());
        token = pref.get(TOKEN_KEY, null);
        username = pref.get(USERNAME_KEY,null);
    }
    static public Authentication getInstance() {
        if (instance == null) {
            instance = new Authentication();
        }
        return instance;
    }
    public String getToken() {
        return token;
    }
    public String getUsername() {
        return username;
    }
    public void saveToken(String token) {
        this.token = token;
        pref.put(TOKEN_KEY, token);
    }
    public void saveUsername(String username) {
        this.username = username;
        pref.put(USERNAME_KEY, username);
    }
    public void deleteToken() {
        this.token = null;
        this.username = null;
        pref.remove(TOKEN_KEY);
        pref.remove(USERNAME_KEY);
    }
}

File: Android/exchange-android-EricNJ/app/src/main/java/com/example/currencyexchange/api/Authentication.kt
package com.example.currencyexchange.api

import android.content.Context
import android.content.SharedPreferences
import android.util.Log

object Authentication {
    private var token: String? = null
    private var preferences: SharedPreferences? = null
    private var username:String?=null
    fun initialize(context: Context) {
        preferences = context.getSharedPreferences("SETTINGS",
            Context.MODE_PRIVATE)
        token = preferences?.getString("TOKEN", null)
        username= preferences?.getString("USERNAME", null)
    }
    fun getToken(): String? {
        return token
    }
    fun saveToken(token: String) {
        this.token = token
        preferences?.edit()?.putString("TOKEN", token)?.apply()
    }
    fun clearToken() {
        this.token = null
        preferences?.edit()?.remove("TOKEN")?.apply()
    }
    fun getUsername(): String? {
        username?.let { Log.e("getus", it) }
        return username
    }
    fun saveUsername(username: String) {
        this.username = username
        preferences?.edit()?.putString("USERNAME",username)?.apply()
    }
    fun clearUsername() {
        this.username = null
        preferences?.edit()?.remove("USERNAME")?.apply()
    }
}
